code42_on_poll_connector.py

```diff
@@ -0,0 +1,287 @@
+                return [ExposureType.not_in(supported_exp_types)]
+        elif self._is_endpoint_exfiltration:
+            return [
+                EventType.is_in(
+                    [EventType.CREATED, EventType.MODIFIED, EventType.READ_BY_APP]
+                ),
+                ExposureType.is_in(exposure_types),
+            ]
+        return []
+
+    def _create_file_category_filters(self):
+        """Determine if file categorization is significant"""
+        observed_file_categories = self._observation_data.get("fileCategories")
+        if observed_file_categories:
+            categories = [
+                get_file_category_value(cat.get("category"))
+                for cat in observed_file_categories
+                if cat.get("isSignificant") and cat.get("category")
+            ]
+            if categories:
+                return FileCategory.is_in(categories)
+
+
+class Code42SearchFilters(object):
+    def __init__(self):
+        self._filters = []
+
+    @property
+    def filters(self):
+        return self._filters
+
+    def to_all_query(self):
+        """Override"""
+
+    def append(self, _filter):
+        if _filter:
+            self._filters.append(_filter)
+
+    def extend(self, _filters):
+        if _filters:
+            self._filters.extend(_filters)
+
+    def append_result(self, value, create_filter):
+        """Safely creates and appends the filter to the working list."""
+        if not value:
+            return
+        _filter = create_filter(value)
+        self.append(_filter)
+
+
+class FileEventQueryFilters(Code42SearchFilters):
+    """Class for simplifying building up a file event search query"""
+
+    def __init__(self, pg_size=None):
+        self._pg_size = pg_size
+        super(FileEventQueryFilters, self).__init__()
+
+    def to_all_query(self):
+        """Convert list of search criteria to *args"""
+        query = FileEventQuery.all(*self._filters)
+        if self._pg_size:
+            query.page_size = self._pg_size
+        return query
+
+
+def _stringify_lists_if_needed(event):
+    # We need to convert certain fields to a stringified list else React.JS will throw an error
+    shared_with = event.get("sharedWith")
+    private_ip_addresses = event.get("privateIpAddresses")
+    if shared_with:
+        shared_list = [
+            u.get("cloudUsername") for u in shared_with if u.get("cloudUsername")
+        ]
+        event["sharedWith"] = str(shared_list)
+    if private_ip_addresses:
+        event["privateIpAddresses"] = str(private_ip_addresses)
+    return event
+
+
+def _create_artifact_json(container_id, alert_details, file_event):
+    event_data = {
+        key: val for key, val in file_event.items() if val not in [[], None, ""]
+    }
+    return {
+        "container_id": container_id,
+        "source_data_identifier": alert_details["id"],
+        "label": alert_details.get("ruleSource"),
+        "cef": event_data,
```

Comment:
Is our event model in CEF format? I remember seeing in a Slack thread that there *exists* a CEF mapping for file events, but does that imply that the raw event data we receive from the `get_file_events` py42 method already puts the file events into CEF?

Reply:
So, it does not really seem like anyone uses the real CEF format here. The most important things is that the data is normalized. Meaning, there is one level and all None values are gone.. Actual CEF fields do not seem to really get used in any of the app I looked at.  That being said, the doc for this field is `A normalized representation of the data mapped to each field's representative CEF key.`. So it is a tad confusing - In the UI, it looks great and so my thought s were to just do it this way unless someone says we shouldn't

Reply:
Update: CEF is the way to go because of how Phantom playbooks reference they keys from artifacts